docs/developer/designs/hierarchical-podgroup/README.md

```diff
@@ -0,0 +1,236 @@
+
+## Examples
+
+### Example 1: PodGroup with Two SubGroups
+A PodGroup with two SubGroups (decode and prefill), each requiring rack-level placement with different minMember constraints.
+```mermaid
+graph TD
+    PG[PodGroup]
+
+    Decode[SubGroup: decoders<br>minMember: 4<br>Topology: rack]
+    Prefill[SubGroup: prefills<br>minMember: 1<br>Topology: rack]
+
+    PG --> Decode
+    PG --> Prefill
+```
+```yaml
+spec:
+  subGroups:
+    - name: decoders
+      podSelector:
+        matchLabels:
+          role: decode
+      minMember: 4
+      topologyConstraints:
+        requiredTopologyLevel: rack
+    - name: prefills
+      podSelector:
+        matchLabels:
+          role: prefill
+      minMember: 1
+      topologyConstraints:
+        requiredTopologyLevel: rack
+```
+
+### Example 2: Multi-SubGroup Leaders and Workers
+A PodGroup with four SubGroups (decode-leaders, decode-workers, prefill-leaders, prefill-workers), enforcing co-location within each group while allowing separation between decode and prefill groups.
+```mermaid
+graph TD
+    PG[PodGroup]
+
+    Decode[SubGroup: decode<br>Topology: block]
+    DecodeWorkers[SubGroup: decode-workers<br>minMember: 4<br>Topology: rack]
+    DecodeLeaders[SubGroup: decode-leaders<br>minMember: 1<br>Topology: rack]
+
+    Prefill[SubGroup: prefill<br>Topology: block]
+    PrefillWorkers[SubGroup: prefill-workers<br>minMember: 4<br>Topology: rack]
+    PrefillLeaders[SubGroup: prefill-leaders<br>minMember: 1<br>Topology: rack]
+
+    PG --> Decode
+    Decode --> DecodeWorkers
+    Decode --> DecodeLeaders
+
+    PG --> Prefill
+    Prefill --> PrefillWorkers
+    Prefill --> PrefillLeaders
+```
+```yaml
+spec:
+  subGroups:
+    - name: decode
+      topologyConstraints:
+        requiredTopologyLevel: block
+    - name: decode-workers
+      parent: decode
+      podSelector:
+        matchLabels:
+          role: decode-worker
+      minMember: 4
+      topologyConstraints:
+        requiredTopologyLevel: rack
+    - name: decode-leaders
+      parent: decode
+      podSelector:
+        matchLabels:
+          role: decode-leader
+      minMember: 1
+      topologyConstraints:
+        requiredTopologyLevel: rack
+
+    - name: prefill
+      topologyConstraints:
+        requiredTopologyLevel: block
+    - name: prefill-workers
+      parent: prefill
+      podSelector:
+        matchLabels:
+          role: prefill-worker
+      minMember: 4
+      topologyConstraints:
+        requiredTopologyLevel: rack
+    - name: prefill-leaders
+      parent: prefill
+      podSelector:
+        matchLabels:
+          role: prefill-leader
+      minMember: 1
+      topologyConstraints:
+        requiredTopologyLevel: rack
+```
+
+## Development
+To ensure a controlled and backward-compatible rollout of PodGroup SubGroups with fine-grained scheduling, the following phased development plan is proposed:
```

Comment:
I don't see here an explanation on how we will keep BC - do we need a conversion webhook? Or some other way to convert old podgroups to the new format?